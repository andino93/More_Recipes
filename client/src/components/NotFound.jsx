import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import FooterComponent from './partials/Footer.jsx';
import NotFoundImage from '../build/assets/images/NFound.gif';

/**
 *
 *
 * @class NotFound
 * @extends {React.Component}
 */
export default class NotFound extends Component {
  /**
   * 
   * @returns {XML} XML/JSX
   * 
   * @memberof NotFound
   */
  render() {
    return (
      <div>
        <nav
          className="navbar
            navbar-expand-lg navbar-light bg-navbar fixed-top">
          <Link to="/" className="navbar-brand" >MoreRecipe</Link>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation">
            <span className="navbar-toggler-icon" />
          </button>
          <div
            className="collapse navbar-collapse justify-content-end"
            id="navbarSupportedContent">
          </div>
        </nav>
        <div className="clearfix mb-20" />
        <div className="container">
          <div className="row">
            <div className="col-12">
              <Link to="/">
                <h2 className="text-center">Go Home</h2>
                <img src={NotFoundImage} style={{ display: 'block', margin: '50px auto' }} alt="Check"/>
              </Link>
            </div>
          </div>
        </div>
        <FooterComponent/>
      </div>
    );
  }
}
