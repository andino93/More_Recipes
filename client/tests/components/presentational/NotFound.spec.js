import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { shallow, configure } from 'enzyme';
import NotFound from '../../../src/components/NotFound.jsx';

configure({ adapter: new Adapter() });

describe('<NotFound>', () => {
  let shallowComponent;
  beforeEach(() => {
    shallowComponent = shallow(<NotFound/>);
  });
  it('renders <NotFound> component', () => {
    const tree = toJson(shallowComponent);
    expect(tree).toMatchSnapshot();
  });
});
