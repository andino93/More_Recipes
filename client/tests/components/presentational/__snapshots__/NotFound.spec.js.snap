// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound> renders <NotFound> component 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-navbar fixed-top"
  >
    <Link
      className="navbar-brand"
      replace={false}
      to="/"
    >
      MoreRecipe
    </Link>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse justify-content-end"
      id="navbarSupportedContent"
    />
  </nav>
  <div
    className="clearfix mb-20"
  />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <Link
          replace={false}
          to="/"
        >
          <h2
            className="text-center"
          >
            Go Home
          </h2>
          <img
            alt="Check"
            src="test-file-stub"
            style={
              Object {
                "display": "block",
                "margin": "50px auto",
              }
            }
          />
        </Link>
      </div>
    </div>
  </div>
  <FooterComponent />
</div>
`;
